src/ui/HierarchicalFacet/HierarchicalFacet.ts

```diff
@@ -93,6 +93,10 @@ interface IFlatHierarchy {
  *
  * Selecting the `folder3` value in the facet would only return the `text2.txt` item.
  *
+ * @deprecated This component is exposed for legacy reasons. You should instead use the {@link CategoryFacet} component. It uses a different
```

Comment:
lol I hate the description that was already there, so I'll edit it all. Here goes:
```
 /**
  * @deprecated This component is exposed for legacy reasons. Instead, use the {@link CategoryFacet} component, which is more performant and easier to use.
  * 
  * The `HierarchicalFacet` component inherits all of its options and behaviors from the [`Facet`]{@link Facet}
  * component, but is meant to be used to render hierarchical values.
  *
  * **Note:**
  * > The `HierarchicalFacet` component does not support the [`customSort`]{@link Facet.options.customSort}
  * > `Facet` option.
  *
  * The `HierarchicalFacet` component can be used to display files in a file system, or categories for items in a
  * hierarchy.
  *
  * This facet requires a group by field with a special format to work correctly.
  *
  * **Example:**
  *
  * You have the following files indexed on a file system:
  * ```
  * c:\
  *   folder1\
  *     text1.txt
  *   folder2\
  *     folder3\
  *       text2.txt
  * ```
  * The `text1.txt` item would have a field with the following format:
  * `c; c|folder1;`
  *
  * The `text2.txt` item would have a field with the following format:
  * `c; c|folder2; c|folder2|folder3;`
  *
  * By default, the `|` character determines the hierarchy (`folder3` inside `folder2` inside `c`).
  *
  * Since both items contain the `c` value, selecting it value in the facet would return both items.
  *
  * Selecting the `folder3` value in the facet would only return the `text2.txt` item.
  * 
  * @notSupportedIn salesforcefree
  */
```